src/MongoDB.Driver.Core/Core/Connections/BinaryConnection.cs

```diff
@@ -921,6 +992,62 @@ public void SentMessages(int bufferLength)
                     handler(new ConnectionSentMessagesEvent(_connection.ConnectionId, _requestIds.Value, bufferLength, networkDuration, _serializationDuration, EventContext.OperationId));
                 }
             }
+
+            // private methods
+            private bool CanMessageBeCompressed(RequestMessage message)
+            {
+                if (IsNotCommandRequestMessage(message))
```

Comment:
Not sure, does this checking make sense? Can we have something different with `CommandRequestMessage` here?

Reply:
I will change on:

`!(message is CommandRequestMessage)`